frontend/src/queries/query.ts

```diff
@@ -96,36 +96,73 @@ async function executeQuery<N extends DataNode>(
      */
     pollOnly = false
 ): Promise<NonNullable<N['response']>> {
-    const isAsyncQuery =
-        methodOptions?.async !== false &&
-        !SYNC_ONLY_QUERY_KINDS.includes(queryNode.kind) &&
-        !!featureFlagLogic.findMounted()?.values.featureFlags?.[FEATURE_FLAGS.QUERY_ASYNC]
+    const isAsyncQuery = methodOptions?.async !== false && !SYNC_ONLY_QUERY_KINDS.includes(queryNode.kind)
 
     const useOptimizedPolling = posthog.isFeatureEnabled('query-optimized-polling')
+    const currentTeamId = teamLogic.findMounted()?.values.currentTeamId
 
     if (!pollOnly) {
         const refreshParam: RefreshType | undefined =
             refresh && isAsyncQuery ? 'force_async' : isAsyncQuery ? 'async' : refresh
-        let response: NonNullable<N['response']>
+
         if (useOptimizedPolling) {
-            response = await api.queryAwaited(
-                queryNode,
-                methodOptions,
-                queryId,
-                refreshParam,
-                filtersOverride,
-                variablesOverride
-            )
-        } else {
-            response = await api.query(
-                queryNode,
-                methodOptions,
-                queryId,
-                refreshParam,
-                filtersOverride,
-                variablesOverride
-            )
+            return new Promise((resolve, reject) => {
+                const abortController = new AbortController()
+
+                void fetchEventSource(`/api/environments/${currentTeamId}/query_awaited/`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                        Accept: '*/*',
```

Comment:
This is alright for now, but definitely needs a comment that it's a hack and the correct value is `text/event-stream`